packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -22,6 +23,38 @@ import 'flutter_command_runner.dart';
 
 typedef void Validator();
 
+enum ExitCode {
+  success,
+  warning,
+  fail,
+}
+
+String _getNullableExitCodeName(ExitCode code) {
+  return code == null ? null : getEnumName(code);
+}
+
+class FlutterCommandResult {
+  FlutterCommandResult(
+    this.exitCode,
+    {
+      this.analyticsParameters,
+      this.exitTime,
+    }
+  ) { assert(exitCode != null); }
+
+  ExitCode exitCode;
+  /// Optional dimension data that can be appended to the timing event.
+  /// https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#timingLabel
+  /// Do not add PII.
+  List<String> analyticsParameters;
+  /// Optional epoch time when the command's non-interactive wait time is 
+  /// complete during the command's execution. Use to measure user perceivable
+  /// latency without measuring user interaction time. 
+  /// 
+  /// Will automatically measure the command's complete time if not provided. 
+  DateTime exitTime;
```

Comment:
No. I think we should measure user perceivable time. That latency measurement should start as early as possible.